src/components/manager/WorkflowListItem.js

```diff
@@ -32,46 +44,41 @@ const ioTagWithType = (id, ioType, typeContent = "") => (
     </Tag>
 );
 
-const WorkflowListItem = ({onClick, workflow}) => {
-    const {inputs, outputs, name, description, data_type: dt} = workflow;
+const FLEX_1 = { flex: 1 };
+const MARGIN_RIGHT_1EM = { marginRight: "1em" };
+
+const WorkflowListItem = ({ onClick, workflow, rightAlignedTags }) => {
+    const {inputs, name, description, data_type: dt} = workflow;
 
     const typeTag = dt ? <Tag key={dt}>{dt}</Tag> : null;
 
     const inputTags = inputs
-        .filter(i => !i.hidden)  // Filter out hidden (often injected/FROM_CONFIG) inputs
-        .map(i => ioTagWithType(i.id, i.type, i.type.startsWith("file") ? i.extensions.join(" / ") : ""));
-
-    const inputExtensions = Object.fromEntries(inputs
-        .filter(i => i.type.startsWith("file"))
-        .map(i => [i.id, i.extensions[0]]));  // TODO: What to do with more than one?
-
-    const outputTags = outputs.map(o => {
-        if (!o.value) console.error("Missing or invalid value prop for workflow output: ", o);
-
-        if (!o.type.startsWith("file")) return ioTagWithType(o.id, o.type);
-
-        const outputValue = o.value || "";
-        let formattedOutput = outputValue;
-
-        [...outputValue.matchAll(/{(.*)}/g)].forEach(([_, id]) => {
-            formattedOutput = formattedOutput.replace(`{${id}}`, {
-                ...inputExtensions,
-                "": o.hasOwnProperty("map_from_input") ? inputExtensions[o.map_from_input] : undefined,
-            }[id]);
-        });
-
-        return ioTagWithType(o.id, o.type, formattedOutput);
-    });
+        .filter(i => !i.hidden && !i.injected)  // Filter out hidden (often injected/FROM_CONFIG) inputs
+        .map(i =>
+            ioTagWithType(
```

Comment:
I think it would make sense for ioTagWithType to be a component that takes a complete workflow input and store its typeContent internally with useMemo.

Reply:
done